### PR TITLE
tgc build on tpg-beta

### PR DESCRIPTION
**Pull request by iyabchen**
terraform-google-conversion switch to tpg beta
- other change is also required in tgc: https://github.com/GoogleCloudPlatform/terraform-google-conversion/pull/1093
- this might requires updating the generate downstream image. 

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Review by melinath — COMMENTED** [review]
could you pull the formatting changes into a separate PR?

**Comment by melinath** [comment]
allowed ci runs /gcbrun

**Review by melinath — CHANGES_REQUESTED** [review]
I think you might need to make additional changes as described in b/275715832

**Review by melinath — CHANGES_REQUESTED** [review]
could you split the downstream-builder changes into a separate PR & also make changes to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/gcb-push-downstream.yml to use beta for TGC

**Comment by iyabchen** [comment]
tgc-base fails because HEAD~ is still ga, not beta. Not sure how to make this work? @melinath

**Comment by melinath** [comment]
testing this will be tricky since it modifies so many files. We'll need to test thoroughly to make sure everything works in the downstream code. I'm seeing some errors when I generate based on this (and switch over to beta) - is everything working fine for you?

File: .ci/gcb-generate-diffs-new.yml [hunk 1]
```diff
@@ -129,7 +129,7 @@ steps:
       args:
           - 'base'
           - 'terraform-google-conversion'
-          - 'ga'
```
Comment (melinath) [inline .ci/gcb-generate-diffs-new.yml:132]: we should temporarily switch this to GA so that the tests can run (even though they'll fail)

**Review by melinath — CHANGES_REQUESTED** [review]
This looks good - my one concern is that it will break older PRs, since the generation will happen with GA set as the version (but downstream uses beta). However, the code that's used for the generation will all be based on the head of the main branch.

In terms of how to work around this - my initial thought is that we could alter the downstream image to always use beta and then deal with any fallout in the ten minutes it takes the image to rebuild manually.

What do you think?

**Comment by melinath** [comment]
@iyabchen could you resolve the conflicts

**Review by melinath — APPROVED** [review]

**Pull request merged**